network/transport/stcp_socket.c

```diff
@@ -0,0 +1,260 @@
+#include <stdlib.h>
+#include <string.h>
+#include "socket.h"
+#include "socket_interface.h"
+#include "error.h"
+#include "util.h"
+#include "trng.h"
+#include "mbedtls/ssl.h"
+#include "mbedtls/entropy.h"
+
+/**
+ * @struct stcp_socket_desc
+ * @brief Fields used by socket
+ */
+struct stcp_sockets_desc {
+	/** TCP Socket reference */
+	struct socket_desc		*sock;
+	/** True random number generator reference */
+	struct trng_desc		*trng;
+	/* Mbed structures */
+	/** Certification authority certificate */
+	mbedtls_x509_crt		ca_cert[1];
+	/** Client certificate */
+	mbedtls_x509_crt		client_cert[1];
+	/** Client private key */
+	mbedtls_pk_context		client_pkey[1];
+	/** SSL configuration strucure */
+	mbedtls_ssl_config		conf[1];
+	/** Mbedtls tls context */
+	mbedtls_ssl_context		ssl[1];
+};
+
+static int32_t stcp_socket_init(struct stcp_socket_desc **desc,
+				struct network_interface *net,
+				struct stcp_socket_init_param *param);
+static int32_t stcp_socket_remove(struct stcp_socket_desc *desc);
+static int32_t stcp_socket_connect(struct stcp_socket_desc *desc,
+				   struct socket_address *addr);
+static int32_t stcp_socket_disconnect(struct stcp_socket_desc *desc);
+static int32_t stcp_socket_send(struct stcp_socket_desc *desc, const void *data,
+				uint32_t len);
+static int32_t stcp_socket_recv(struct stcp_socket_desc *desc, void *data,
+				uint32_t len);
+
+/* Secure socket interface */
+const struct socket_interface stcp_socket_interface = {
+	.protocol = PROTOCOL_TCP,
+	/* Will be initialized in socket.c */
+	.sock = NULL,
+	.init = (int32_t (*)())stcp_socket_init,
```

Comment:
is this cast necessary ? I would remove it.

Reply:
Yes, to avoid the warning